tests/Cookie/CookieJarTest.php

```diff
@@ -37,6 +37,19 @@ public function testCreatesFromArray()
         $this->assertCount(2, $jar);
     }
 
+    public function testGetsCookiesByName()
+    {
+        $cookies = $this->getTestCookies();
+        foreach ($this->getTestCookies() as $cookie) {
+            $this->jar->setCookie($cookie);
+        }
+
+        $testCookie = $cookies[0];
+        $this->assertEquals($testCookie, $this->jar->getCookieByName($testCookie->getName()));
+        $this->assertNull($this->jar->getCookieByName("doesnotexist"));
+        $this->assertNull($this->jar->getCookieByName(null));
```

Comment:
Is this a valid usage of this function? Even if the cookie name is `null` we would never return it with the current implementation. 

See my suggestion above.